kaskus/pecl-mogilefs
# PHP 8 compatibility

### PHP 8 Compatibility + backward compatibility for PHP 7
- rebuild arginfo for PHP 8 with __construct
- separate arginfo file for PHP 8 (`mogilefs_arginfo.h`) and PHP 7 (`mogilefs_legacy_arginfo.h`)
- change type_hint "**integer**" to "**int**" in some methods (using integer triggers warning)
https://github.com/kaskus/pecl-mogilefs/blob/21cb55eee9be83d865e9daf828181731ccb1c541/mogilefs.c#L88-L91

Warning triggered when run unit test on general repository

```
PHP Warning:  "integer" will be interpreted as a class name. Did you mean "int"? Write "\integer" to suppress this warning in phar:///home/kaskus/general/vendor/kaskus/kaskus-phar/archive/phpunit/phpunit/Framework/MockObject/MockClass.php(48) : eval()'d code on line 9
```
- update test for PHP 8
- move old test to `*_7.php` for PHP 7

How to test (both on PHP 8 and PHP 7)

1. Checkout this branch
2. run phpize in root folder
3. run command `./configure --with-php-config=/path/php-config`
4. copy `tests/config.inc.php.example` to  `tests/config.inc.php`. Change the value depending on your local environment
5. run command `make test`

You will see output like this. All tests must passed ( no failed test )

![Screen Shot 2021-11-17 at 4 51 14 PM](https://user-images.githubusercontent.com/2119356/142177708-4bf8ab88-d493-4683-91a7-2ffbfb4d31f0.png)


```
=====================================================================
TEST RESULT SUMMARY
---------------------------------------------------------------------
Exts skipped    :    0
Exts tested     :   40
---------------------------------------------------------------------

Number of tests :   26                17
Tests skipped   :    9 ( 34.6%) --------
Tests warned    :    0 (  0.0%) (  0.0%)
Tests failed    :    0 (  0.0%) (  0.0%)
Tests passed    :   17 ( 65.4%) (100.0%)
---------------------------------------------------------------------
Time taken      :    3 seconds
=====================================================================
```


Tested on : 
1. PHP 8.0.10
2. PHP 7.4.3
